specs/commandline/certificate.md

```diff
@@ -0,0 +1,220 @@
+                    cert-3.pem   # certs under sub directory is ignored
+
+        /signingAuthority
+            /wabbit-networks
+                cert3.crt
+```
+
+In this example, there are two certificates stored in trust store named `acme-rockets` of type `ca`. There is one certificate stored in trust store named `wabbit-networks` of type `signingAuthority`.
+
+## Outline
+
+### notation certificate
+
+```text
+Manage certificates in trust store for signature verification.
+
+Usage:
+  notation certificate [command]
+
+Aliases:
+  certificate, cert
+
+Available Commands:
+  add           Add certificates to the trust store.
+  delete        Delete certificates from the trust store.
+  generate-test Generate a test RSA key and a corresponding self-signed certificate.
+  list          List certificates in the trust store.
+  show          Show certificate details given trust store type, named store, and certificate file name. If the certificate file contains multiple certificates, then all certificates are displayed.
+
+Flags:
+  -h, --help   help for certificate
+```
+
+### notation certificate add
+
+```text
+Add certificates to the trust store.
+
+Usage:
+  notation certificate add --type <type> --store <name> [flags] <cert_path>...
+
+Flags:
+  -h, --help           help for add
+  -s, --store string   specify named store
+  -t, --type string    specify trust store type, options: ca, signingAuthority
+```
+
+### notation certificate list
+
+```text
+List certificates in the trust store.
+
+Usage:
+  notation certificate list [flags]
+
+Aliases:
+  list, ls
+
+Flags:
+  -h, --help           help for list
+  -s, --store string   specify named store
+  -t, --type string    specify trust store type, options: ca, signingAuthority
+```
+
+### notation certificate show
+
+```text
+Show certificate details given trust store type, named store, and certificate file name. If the certificate file contains multiple certificates, then all certificates are displayed.
+
+Usage:
+  notation certificate show --type <type> --store <name> [flags] <cert_fileName>
+
+Flags:
+  -h, --help           help for show
+  -s, --store string   specify named store
+  -t, --type string    specify trust store type, options: ca, signingAuthority
+```
+
+### notation certificate delete
+
+```text
+Delete certificates from the trust store.
+
+Usage:
+  notation certificate delete --type <type> --store <name> [flags] (--all | <cert_fileName>)
+
+Aliases:
+  delete, rm
+
+Flags:
+  -a, --all            remove all certificates in the named store
+  -h, --help           help for delete
+  -s, --store string   specify named store
+  -t, --type string    specify trust store type, options: ca, signingAuthority
+  -y, --yes            do not prompt for confirmation
+```
+
+### notation certificate generate-test
+
+```text
+Generate a test RSA key and a corresponding self-signed certificate.
+
+Usage:
+  notation certificate generate-test [flags] <host>
```

Comment:
instead of `host`, can we just use `name` and remove `--name` flag? In code we are using host as common name of certificate Ref: https://github.com/notaryproject/notation-core-go/blob/b5df54c7465b0c1b177233108355b4e7314f1a14/testhelper/certificatetest.go#L166

Reply:
Yes but how will this host information be useful? Is it more kind of a common-name for certificate? If so we should remove host and use name or common-name